src/resonate/actions.py

```diff
@@ -59,3 +68,180 @@ def with_options(
 @dataclass(frozen=True)
 class Sleep:
     seconds: int
+
+
+class Combinator(Generic[T]):
+    opts: Options = field(default=Options())
+
+    def __init__(self) -> None:
+        self.opts = Options(retry_policy=never())
+
+    def with_options(
+        self,
+        *,
+        durable: bool = True,
+        promise_id: str | None = None,
+        retry_policy: RetryPolicy | None = None,
+    ) -> Self:
+        """
+        Set options for the combinator.
+
+        Args:
+            durable (bool): Whether the promise is durable. Defaults to True.
+            promise_id (str | None): An optional identifier for the promise.
+            retry_policy (RetryPolicy | None): An optional retry policy for the promise.
+
+        Returns:
+            Self: The combinator instance with updated options.
+        """
+        self.opts = Options(
+            durable=durable,
+            promise_id=promise_id,
+            retry_policy=retry_policy if retry_policy is not None else never(),
+        )
+        return self
+
+    @abstractmethod
+    def done(self) -> bool:
+        pass
+
+    @abstractmethod
+    def result(self) -> Result[T, Exception]:
```

Comment:
All methods in the `actions` classes are meant to be used by the user. I don't really love the idea of having `result()` and `done()` here. It seems confusing to me, I don't know what you think

Reply:
Where should we put them? should them be private methods of the scheduler?

Reply:
It believe so. This methods belong to the kernel space, and shouldn't face the user.